starter-code/basic-algorithms.js

```diff
@@ -1,7 +1,155 @@
-// Names and Input
+// 1. Create a variable `hacker1` with the driver's name
 
+var hacker1 = "Samuel"; // driver
 
-//Conditionals
+// 2. Print `"The driver's name is XXXX"` 
+console.log("The driver's name is " + hacker1);
 
+// 3. Create a variable `hacker2` and [ask the user
+var hacker2 = prompt("What's your name ?"); // navigator
 
-// Lorem ipsum generator
+// 4. Print `"The navigator's name is YYYY"`
+console.log("The navigator's name is " + hacker2);
+
+//##
+/*
+5. Depending on which name is longer
+	- `The Driver has the longest name, it has XX characters` or 
+	- `Yo, navigator got the longest name, it has XX characters` or
+	- `wow, you both got equally long names, XX characters!!`
+*/
+
+if(hacker1.length > hacker2.length) {
+	console.log(hacker1 + " has the longest name." );
+} else if(hacker1.length < hacker2.length) {
+	console.log(hacker2 + " has the longest name.");
+} else {
+	console.log("you both got equally long names !");
+}
+
+//## Loops
+/*
+6. Print all the characters of the driver's name, separated by a space and [in capitals]
+ "J O H N"`
+  */
+
+var i = 0;
+var result = "";
+
+while(i < hacker1.length ) {	
```

Comment:
While this works, It's better to use a for loop because you're less likely to forget things. 

An alternative way to make everything uppercase, is to just do it once at the end on the final result. `result.toUpperCase()`